larray/core/array.py

```diff
@@ -6332,34 +6337,66 @@ def combine_axes(self, axes=None, sep='_', wildcard=False):
         >>> arr.combine_axes(sep='/')
         a/b  a0/b0  a0/b1  a0/b2  a1/b0  a1/b1  a1/b2
                  0      1      2      3      4      5
-        >>> arr = ndtest((2, 3, 4))
+        >>> arr = ndtest((2, 2, 2, 2))
         >>> arr
-         a  b\\c  c0  c1  c2  c3
-        a0   b0   0   1   2   3
-        a0   b1   4   5   6   7
-        a0   b2   8   9  10  11
-        a1   b0  12  13  14  15
-        a1   b1  16  17  18  19
-        a1   b2  20  21  22  23
-        >>> arr.combine_axes((x.a, x.c))
-        a_c\\b  b0  b1  b2
-        a0_c0   0   4   8
-        a0_c1   1   5   9
-        a0_c2   2   6  10
-        a0_c3   3   7  11
-        a1_c0  12  16  20
-        a1_c1  13  17  21
-        a1_c2  14  18  22
-        a1_c3  15  19  23
+         a   b  c\\d  d0  d1
+        a0  b0   c0   0   1
+        a0  b0   c1   2   3
+        a0  b1   c0   4   5
+        a0  b1   c1   6   7
+        a1  b0   c0   8   9
+        a1  b0   c1  10  11
+        a1  b1   c0  12  13
+        a1  b1   c1  14  15
+        >>> arr.combine_axes(('a', 'c'))
+          a_c  b\\d  d0  d1
+        a0_c0   b0   0   1
+        a0_c0   b1   4   5
+        a0_c1   b0   2   3
+        a0_c1   b1   6   7
+        a1_c0   b0   8   9
+        a1_c0   b1  12  13
+        a1_c1   b0  10  11
+        a1_c1   b1  14  15
+        >>> arr.combine_axes(('a', 'c'), name='ac')
+           ac  b\\d  d0  d1
+        a0_c0   b0   0   1
+        a0_c0   b1   4   5
+        a0_c1   b0   2   3
+        a0_c1   b1   6   7
+        a1_c0   b0   8   9
+        a1_c0   b1  12  13
+        a1_c1   b0  10  11
+        a1_c1   b1  14  15
+        >>> arr.combine_axes({('a', 'c'): 'ac', ('b', 'd'): None})
```

Comment:
I wonder if we shouldn't support this instead (or in addition to the dict syntax):
```python
>>> arr.combine_axes([('a', 'c'), ('b', 'd')])
```
because the None is kinda annoying to type.

Reply:
I agree to implement this:
```python
>>> arr.combine_axes('a,c;b,d')
>>> arr.combine_axes('a,c;b,d', names='ac;bd')

>>> arr.combine_axes([('a', 'c'), ('b', 'd')])
>>> arr.combine_axes([('a', 'c'), ('b', 'd')], names=['ac', 'bd'])
```
If users want to use `names`, they should provide all names.

Reply:
I wonder if requiring the dict syntax when the user wants to provide names is not an option:

We would have:
```python
>>> arr.combine_axes(('a', 'c'))
>>> arr.combine_axes({('a', 'c'): 'ac'})
>>> arr.combine_axes([('a', 'c'), ('b', 'd')])
>>> arr.combine_axes({('a', 'c'): 'ac', ('b', 'd'): 'bd'})
```

Reply:
dict are not ordered for Python <= 3.5. This is why I have a loop in `LArray.combine_axes` to compute "transposed_axes". I would prefer to avoid dict.

Reply:
OK